### PR TITLE
Bulletproof AbstractProxyServlet#destory() to make it easier to write

### PR DESCRIPTION
**Pull request by garydgregory**
unit tests for custom subclasses of AbstractProxyServlet

### CODE REVIEW

**Comment by garydgregory** [comment]
Hi @olamy 
This PR is based on the 10.0.x branch.

File: jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java [hunk 1]
```diff
@@ -145,11 +145,14 @@ public void destroy()
     {
         try
         {
-            _client.stop();
+            if (_client != null)
+                _client.stop();
```
Comment (gregw) [inline jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java]: Use:
```suggestion
            LifeCycle.stop(_client);
```
As it is more concise and also avoids any race conditions if _client changes value.

File: jetty-proxy/src/test/java/org/eclipse/jetty/proxy/AsyncMiddleManServletTest.java [hunk 2]
```diff
@@ -159,9 +159,18 @@ private void startClient() throws Exception
...
     {
-        client.stop();
-        proxy.stop();
-        server.stop();
+        if (client != null)
+            client.stop();
+        if (proxy != null)
+            proxy.stop();
+        if (server != null)
+            server.stop();
```
Comment (gregw) [inline jetty-proxy/src/test/java/org/eclipse/jetty/proxy/AsyncMiddleManServletTest.java]:
```suggestion
        LifeCycle.stop(client);
        LifeCycle.stop(proxy);
        LifeCycle.stop(server);
```

**Review by gregw — CHANGES_REQUESTED** [review]
just some small improvements

File: jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java (hunk 1 above)
Comment (garydgregory) [reply jetty-proxy/src/main/java/org/eclipse/jetty/proxy/AbstractProxyServlet.java]: @gregw 👍 Thank you for the review. Done.

File: jetty-proxy/src/test/java/org/eclipse/jetty/proxy/AsyncMiddleManServletTest.java (hunk 2 above)
Comment (garydgregory) [reply jetty-proxy/src/test/java/org/eclipse/jetty/proxy/AsyncMiddleManServletTest.java]: @gregw 👍 Thank you for the review. Done.

**Review by olamy — APPROVED** [review]

**Review by gregw — APPROVED** [review]

**Comment by gregw** [comment]
Hmm some CI failure.  re-running the build....

**Comment by gregw** [comment]
@olamy if you are still around, any idea what is going on with CI for this one? It passes JDK11 & 19, but fails JDK17?

**Pull request merged**